psycopg/psycopg/rows.py

```diff
@@ -211,6 +211,32 @@ def kwargs_row__(values: Sequence[Any]) -> T:
     return kwargs_row_
 
 
+def scalar_row(index: int = 0) -> BaseRowFactory[Any]:
+    """Generate a row factory returning a column at `!index` as a scalar value.
+
+    :param index: The index of a column to return as a scalar value.
+    """
+
+    def scalar_row_(cursor: "BaseCursor[Any, Any]") -> "RowMaker[Any]":
+        res = cursor.pgresult
+        if not res:
+            return no_result
+    
+        nfields = _get_nfields(res)
+        if nfields is None:
+            return no_result
+
+        if index >= nfields:
+            return no_result
```

Comment:
`no_result` raises an `InterfaceError("the cursor doesn't have a result")` so it doesn't seem appropriate here. It would be more appropriate to raise a `ProgrammingError` if the user asked for a non-existing column. It would be also expected to handle negative numbers as per normal Python indexing and raise the same exception if the index is < -nfields.

Reply:
I will check, if the cursor contains exactly one column, if not, i will raise `ProgrammingError`.